src/app/static/src/tests/adrUpload/mutations.test.ts

```diff
@@ -56,4 +59,43 @@ describe("ADR mutations", () => {
         expect(state.currentFileUploading).toBe(null);
         expect(state.totalFilesUploading).toBe(null);
     });
+
+    it("can clear status", () => {
+        const state = mockADRUploadState();
+        mutations[ADRUploadMutation.ClearStatus](state);
+        expect(state.uploadComplete).toBe(false);
+        expect(state.releaseCreated).toBe(false);
+        expect(state.releaseFailed).toBe(false);
+    });
+
+    it("can set release created", () => {
+        const state = mockADRUploadState();
+        mutations[ADRUploadMutation.ReleaseCreated](state);
+        expect(state.releaseCreated).toBe(true);
+    });
+
+    it("can set release failed with error", () => {
+        const state = mockADRUploadState();
+        mutations[ADRUploadMutation.ReleaseFailed](state, {payload: mockError("Version already exists for this activity")});
+        expect(state.releaseFailed).toBe(true);
+        expect(state.uploadError!!.detail).toBe("A release already exists on ADR for the latest files");
+
+        mutations[ADRUploadMutation.ReleaseFailed](state, {payload: mockError("Version names must be unique per dataset")});
+        expect(state.releaseFailed).toBe(true);
+        expect(state.uploadError!!.detail).toBe("Release names must be unique per dataset and a release with the same name already exists on ADR. Try renaming the project to generate a new release name.");
+
+        mutations[ADRUploadMutation.ReleaseFailed](state, {payload: mockError("other error")});
+        expect(state.releaseFailed).toBe(true);
+        expect(state.uploadError!!.detail).toBe("other error");
+    });
+
+    it("can set release failed with error in French", () => {
+        const state = mockADRUploadState();
+        i18next.changeLanguage("fr");
```

Comment:
This returns a promise, so really should have an `await`:
```suggestion
        await i18next.changeLanguage("fr");
```